src/DrTests-TestCoverage/DTTestCoverageResult.class.st

```diff
@@ -10,13 +10,33 @@ Class {
 	#instVars : [
 		'methodList',
 		'percent',
-		'partiallyCoveredMethods'
+		'partiallyCoveredMethods',
+		'coveredMethods'
 	],
 	#category : #'DrTests-TestCoverage'
 }
 
+{ #category : #'tree building' }
+DTTestCoverageResult >> buildTreeByClass: aMethodList [
+
+	| methodsByClasses subResultsTree |
+	methodsByClasses := self methodsGroupedByClass: aMethodList.
+	subResultsTree := {  }.
+	methodsByClasses keysAndValuesDo: [ :key :value |
+		| classNode |
+		classNode := DTTreeNode new
+			             name: key asString;
+			             subResults:
+				             (value collect: [ :each |
+						              each asResultForDrTestCoverage ]);
+			             yourself.
+		subResultsTree := subResultsTree , { classNode } ].
+	^ subResultsTree
```

Comment:
There is already an existing method to group collection elements called `groupedBy:`.
It is more efficient to create an OrderedCollection, add elements to it then finally convert it into an array, rather than creating new arrays repeatedly (which `,` does).
```suggestion
	| subResultsTree |
	subResultsTree := OrderedCollection new.
	(aMethodList groupedBy: #methodClass) keysAndValuesDo: [ :key :value |
		subResultsTree add: (DTTreeNode new
				 name: key asString;
				 subResults:
					 (value collect: [ :each | DTTestCoverageLeafNode content: each ])) ].
	^ subResultsTree asArray
```